src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -668,8 +668,12 @@ InterpreterCreateQuery::TableProperties InterpreterCreateQuery::getTableProperti
 
         if (create.columns_list->indices)
             for (const auto & index : create.columns_list->indices->children)
+            {
                 properties.indices.push_back(
                     IndexDescription::getIndexFromAST(index->clone(), properties.columns, getContext()));
+                    if (properties.indices[properties.indices.size()-1].type == "annoy" && !getContext()->getSettingsRef().allow_experimental_annoy_index)
```

Comment:
Maybe `if (properties.indices.back().type [...]`?

Reply:
And just for curiosity: Did you test that case-insensitive variations of "annoy" (e.g. "Annoy") are also blocked?